controllers/openstackdataplane_controller.go

```diff
@@ -153,44 +141,45 @@ func (r *OpenStackDataPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 			labels := client.MatchingLabels(labelSelector)
 			listOpts = append(listOpts, labels)
 		}
-		err = r.Client.List(ctx, roles, listOpts...)
+
+		err = r.Client.List(ctx, nodeSets, listOpts...)
 		if err != nil {
 			return ctrl.Result{}, err
 		}
-		r.Log.Info("found roles", "total", len(roles.Items))
-		if len(instance.Spec.Roles) > len(roles.Items) {
-			shouldRequeue = true
-		}
 
-		for _, role := range roles.Items {
-			logger.Info("DataPlane deploy", "role.Name", role.Name)
-			if role.Spec.DataPlane != instance.Name {
-				err = fmt.Errorf("role %s: role.DataPlane does not match with role.Label", role.Name)
-				deployErrors = append(deployErrors, "role.Name: "+role.Name+" error: "+err.Error())
+		for _, nodeSet := range nodeSets.Items {
+			logger.Info("DataPlane deploy", "nodeSet.Name", nodeSet.Name)
+			// We don't expect that our nodeSet list will be overly large. As such, linear search is likely
+			// to be computationally more efficient compared to importing and using slices.Contains() here.
+			nodeSetFound := false
+			for _, nodeSetName := range instance.Spec.NodeSets {
+				if nodeSetName == nodeSet.Name {
+					nodeSetFound = true
+				}
 			}
-			r.Log.Info("Role", "DeployStrategy.Deploy", role.Spec.DeployStrategy.Deploy, "Role.Namespace", instance.Namespace, "Role.Name", role.Name)
-			if !role.Spec.DeployStrategy.Deploy {
```

Comment:
I thought we had agreed to leave the DeployStrategy changes to a separate PR?

Reply:
Reverted here. I would like to have some way of triggering based on the existence of the `OpenStackDataPlane` CR though. So that controller needs some mechanism to trigger this deployment. Otherwise we're still in a state where the CR's can be created multiple different ways and the existence of the OpenStackDataPlane controller is redundant.

And maybe the answer is that all of this deploy logic is moved into the OpenStackDataPlane controller instead of here? Not too sure, but using the label was just the first thing that come to mind without editing the spec section of the user created object.